src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -3565,14 +3565,13 @@ public override SyntaxNode NameOfExpression(SyntaxNode expression)
         }
 
         public override SyntaxNode ReturnStatement(SyntaxNode expressionOpt = null)
-        {
-            return SyntaxFactory.ReturnStatement((ExpressionSyntax)expressionOpt);
-        }
+            => SyntaxFactory.ReturnStatement((ExpressionSyntax)expressionOpt);
+
+        internal override SyntaxNode YieldReturnStatement(SyntaxNode expressionOpt = null)
```

Comment:
public?

Reply:
Possibly.  I would prefer to not have to go through all that rigamarole.  We could certainly do a public-API pass here later if people wanted these guys :)

Reply:
If you make it public, please also add generator unit tests.

Reply:
I'll add tests even as internal.  It would be good to test at the low level as well!

Reply:
tests added.  tahnks!